### PR TITLE
Create initial page for `CS8422`

### PR DESCRIPTION
**Pull request by BartoszKlonowski**
This pull request introduces the page describing `CS8422` error message,
with it's reasoning (mostly based on dotnet/csharplang#1277), scenarios and possible correction.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/compiler-messages/cs8422.md](https://github.com/dotnet/docs/blob/0d2d1a87e1f6fe12633b0a2767557eb79b58d1a5/docs/csharp/language-reference/compiler-messages/cs8422.md) | [Compiler Error CS8422](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs8422?branch=pr-en-us-38813) |
| [docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md](https://github.com/dotnet/docs/blob/0d2d1a87e1f6fe12633b0a2767557eb79b58d1a5/docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md) | [Sorry, we don't have specifics on this C# error](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error?branch=pr-en-us-38813) |

### CODE REVIEW

**Comment by BillWagner** [comment]
I'm about to review the main text, but I see the TOC hasn't been updated yet. Can you make that commit @BartoszKlonowski ?

File: docs/csharp/language-reference/compiler-messages/cs8422.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+f1_keywords:
+  - "CS8422"
+helpviewer_keywords:
+  - "CS8422"
+---
+# Compiler Error CS8422
+
+A static local function cannot contain a reference to 'this' or 'base'.
+
+As of C# 8.0 [`static`](../keywords/static.md) keyword is allowed to be used with local functions to prevent from accidentially shadowing or anyhow modifying the state of parent methods or instances fields.
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8422.md:14]: a couple wording nits:

```suggestion
The [`static`](../keywords/static.md) keyword on a local function prevents the local function from accessing the state of parent instance methods or instances fields.
```

File: docs/csharp/language-reference/compiler-messages/cs8422.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+<br/>This error is to indicate about such scenario, of mixing contexts, taking place.
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8422.md:15]: Change to present tense: and active voice:

```suggestion
<br/>This error indicates the local function accesses instance members of the containing type.
```

File: docs/csharp/language-reference/compiler-messages/cs8422.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        localFunc(10);
+        Console.WriteLine(this.counter);
+    }
+}
+```
+
+## To correct this error
+
+If there is an intention of local function to capture and modify the state of a parent then it shouldn't be declared static:
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8422.md:44]: I'd also add a sentence that you could remove the access to instance members:

```suggestion
If there is an intention of local function to capture and modify the state of a parent then it shouldn't be declared static. You could also remove access to any instance members.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Hi @BartoszKlonowski 

I have a couple more wording nits as I reviewed the text. Once we work on those, this is ready to merge.

**Comment by BartoszKlonowski** [comment]
> I'm about to review the main text, but I see the TOC hasn't been updated yet. Can you make that commit @BartoszKlonowski ?

@BillWagner What about commit https://github.com/dotnet/docs/pull/38813/commits/7c3dc7061567b9c61ddaa09060505e37018f0282? Is that toc.ml not the right one?

**Comment by BillWagner** [comment]
> @BillWagner What about commit [7c3dc70](https://github.com/dotnet/docs/commit/7c3dc7061567b9c61ddaa09060505e37018f0282)? Is that toc.ml not the right one?

You're right. I was looking at the list of files in the preview table, not in the files tab. It was my mistake.

**Comment by BartoszKlonowski** [comment]
@BillWagner Thank you for the suggestions! I applied them.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @BartoszKlonowski 

I'll merge this as soon as the build finishes.

**Pull request merged**